src/test/java/com/amazon/ion/impl/EncodingDirectiveCompilationTest.java

```diff
@@ -187,10 +203,32 @@ IonRawWriter_1_1 newWriter(OutputStream out) {
             EncodingContext getEncodingContext(IonReader reader) {
                 return ((IonReaderTextSystemX) reader).getEncodingContext();
             }
+
+            @Override
+            Map<MacroRef, Macro> newMacroTableByMacroRef(SortedMap<String, Macro> macrosByName) {
+                Map<MacroRef, Macro> macroTable = new HashMap<>();
+                int id = 0;
+                for (Map.Entry<String, Macro> nameAndMacro : macrosByName.entrySet()) {
+                    Macro macro = nameAndMacro.getValue();
+                    macroTable.put(MacroRef.byId(id++), macro);
+                    String name = nameAndMacro.getKey();
+                    if (name != null) {
+                        macroTable.put(MacroRef.byName(name), macro);
+                    }
+                }
+                return macroTable;
+            }
+
+            @Override
+            void startMacroInvocationByName(IonRawWriter_1_1 writer, String name, Map<String, Macro> macrosByName) {
```

Comment:
In the tests, sometimes we will invoke a macro by name in text, and other times by ID, to provide adequate coverage for both cases.